libethdrivers/src/helpers.c

```diff
@@ -1,33 +1,41 @@
 /*
  * Copyright 2017, Data61, CSIRO (ABN 41 687 119 230)
+ * Copyright 2020, HENSOLDT Cyber GmbH
  *
  * SPDX-License-Identifier: GPL-2.0-only
  */
 
 #include <ethdrivers/helpers.h>
 
-dma_addr_t
-dma_alloc_pin(ps_dma_man_t *dma_man, size_t size, int cached, int alignment)
+dma_addr_t dma_alloc_pin(ps_dma_man_t *dma_man, size_t size, int cached, int alignment)
 {
     void *virt = ps_dma_alloc(dma_man, size, alignment, cached, PS_MEM_NORMAL);
     if (!virt) {
-        return (dma_addr_t) {0, 0};
+        LOG_ERROR("ps_dma_alloc() failed for size=%zu, alignment=%d",
```

Comment:
These should use `ZF_LOGE` in any new code to be consistent. The only LOG_ERROR macros are from a time when every library invented it's own logging macros and are deprecated.

Reply:
I will change this